### PR TITLE
Feature/revenue sharing

### PR DESCRIPTION
**Pull request by arunjot12**
Added new feature of revenue sharing in which the contract deployer will receive 50% of the caller transaction fees

### CODE REVIEW

File: frame/transaction-payment/src/payment.rs [hunk 1]
```diff
@@ -140,11 +140,12 @@ where
 			let mut refund_owner = C::PositiveImbalance::zero();
 			let contract_address = ContractDeployer::<T>::get(who);
 			if let Some(contract_address) = contract_address {
-				let half_fee = corrected_fee / 2u32.into();
+				let deployer_fee = corrected_fee / 2u32.into();
```
Comment (s-h-ubham) [inline frame/transaction-payment/src/payment.rs:143]: @arunjot12 change name of deployer_fee to refund_fee

**Review by s-h-ubham — CHANGES_REQUESTED** [review]

File: frame/transaction-payment/src/payment.rs (hunk 1 above)
Comment (arunjot12) [reply frame/transaction-payment/src/payment.rs]: @s-h-ubham Please check

**Review by s-h-ubham — APPROVED** [review]

File: frame/evm/src/lib.rs [hunk 2]
```diff
@@ -940,17 +943,34 @@ where
...
 		base_fee: U256,
 		already_withdrawn: Self::LiquidityInfo,
+		target: Option<H160>
 	) -> Self::LiquidityInfo {
 		if let Some(paid) = already_withdrawn {
 			let account_id = T::AddressMapping::into_account_id(*who);
 
 			// Calculate how much refund we should return
 			let refund_amount = paid.peek().saturating_sub(corrected_fee.unique_saturated_into());
+			let caller_fees = corrected_fee / 2;
```
Comment (dung5ire) [inline frame/evm/src/lib.rs]: Can we add some comments here? Our purpose that sharing 50% fee for contract deployer from the caller transaction fee

File: frame/evm/src/lib.rs [hunk 3]
```diff
@@ -554,6 +554,8 @@ pub mod pallet {
 			}
 		}
 	}
+	#[pallet::storage]
+	pub type ContractDeployer<T: Config> = StorageMap<_, Blake2_128Concat, H160, H160>;
```
Comment (dung5ire) [inline frame/evm/src/lib.rs:561]: give some comments , what is key, value ?

**Review by dung5ire — CHANGES_REQUESTED** [review]

File: frame/evm/src/lib.rs (hunk 2 above)
Comment (arunjot12) [reply frame/evm/src/lib.rs]: @dung5ire Please check

File: frame/evm/src/runner/stack.rs [hunk 4]
```diff
@@ -480,7 +487,14 @@ where
 					executor.transact_create(source, value, init, gas_limit, access_list);
 				(reason, address)
 			},
-		)
+		);
+		let execute_result = execute.as_ref();
+		let value = execute_result.unwrap().value;
```
[the remark this answers is not in the document]
Comment (dung5ire) [reply frame/evm/src/runner/stack.rs:492]: can we give 1 comment for it ?

File: frame/evm/src/runner/stack.rs [hunk 5]
```diff
@@ -480,7 +487,14 @@ where
...
+		let owner = ContractDeployer::<T>::get(value);
```
Comment (dung5ire) [inline frame/evm/src/runner/stack.rs:493]: can we have 1 comment for it?

File: frame/evm/src/runner/stack.rs [hunk 6]
```diff
@@ -540,7 +555,14 @@ where
 					executor.transact_create2(source, value, init, salt, gas_limit, access_list);
 				(reason, address)
 			},
-		)
+		);
+		let execute_result = execute.as_ref();
+		let value = execute_result.unwrap().value;
```
Comment (dung5ire) [inline frame/evm/src/runner/stack.rs:560]: the same above

File: frame/evm/src/runner/stack.rs [hunk 7]
```diff
@@ -540,7 +555,14 @@ where
...
+		let owner = ContractDeployer::<T>::get(value);
```
Comment (dung5ire) [inline frame/evm/src/runner/stack.rs:561]: the same above

**Review by dung5ire — CHANGES_REQUESTED** [review]

File: frame/evm/src/runner/stack.rs (hunk 7 above)
Comment (arunjot12) [reply frame/evm/src/runner/stack.rs]: Added. Please check

File: frame/evm/src/runner/stack.rs (hunk 6 above)
Comment (arunjot12) [reply frame/evm/src/runner/stack.rs:566]: Done

File: frame/evm/src/runner/stack.rs (hunk 5 above)
Comment (arunjot12) [reply frame/evm/src/runner/stack.rs]: Done

**Review by s-h-ubham — APPROVED** [review]

**Pull request merged**